publication/ver11/cr/security_and_privacy.md

```diff
@@ -0,0 +1,450 @@
+  a TD Consumer to reliably connect to a described Thing.
+* Profiles define a subset of all TDs, so the information provided is also a subset.
+* Discovery provides a set of TDs, and additional metadata for managing TDs, such as time of last update. 
+  This additional metadata is necessary to identify stale information.
+
+### 3 [How do the features in your specification deal with personal information, personally-identifiable information (PII), or information derived from them?](https://www.w3.org/TR/security-privacy-questionnaire/#personal-data)
+Currently WoT specifications do not deal directly with sensor data but with metadata.
+The WoT Thing Description metadata describes IoT devices and may include
+information such as location and type.  If these devices can be associated
+with an owner, then it may be possible to infer information about the
+owner.  For example, if a user is associated with a device as an owner,
+and that device is a baby monitor, then the user may have a baby,
+the user is probably in a certain age range, and so forth.
+
+The WoT Thing Description (TD) also allows extension via the inclusion
+of custom vocabularies.  Although the WoT standard itself does not
+have any explicit requirement for PII, it is possible an extension might.  
+
+As for the indirect risk, in our Security Considerations
+we recommend that as a precaution a 
+WoT Thing Description should be treated _as if_ it contained PII
+and be stored, cached, and transmitted accordingly.  For example,
+we recommend that TDs only be distributed to authorized and authenticated
+users and only be cached for a limited time.
+
+Since we assume that TDs can contain PII or that PII can be inferred from them
+in some circumstances, the Discovery specification
+includes features to address these points.
+In particular, TD Directories (part of the Discovery specification)
+support access control (so that TDs can be limited in distribution to authorized
+parties) and expiry requirements (so that TDs will be automatically removed when
+they expire) and mechanisms to explicitly delete registered TDs.
+Access controls are based on known best methods for web services (OAuth, etc).
+Other security and privacy considerations warn against embedding metadata in
+public information that is not protected,
+such as URLs presented during Discovery's Introduction phase.
+
+### 4 [How do the features in your specification deal with sensitive information?](https://www.w3.org/TR/security-privacy-questionnaire/#sensitive-data)
+Sensitive information can include both security secrets (which should never
+be distributed to other parties) and information that may be distributed
+but only under specific conditions. 
+ 
+A WoT Thing may both require credentials to be accessed and
+use credentials to access other devices.  However, these are generally
+to secure M2M communication and are not (or should not be) tied directly to
+user credentials used for other services.
+
```

Comment:
This statement might be wrong. Why it should be different? A WoT Thing needs both require credentials no matter which/whom the Thing communicate.

Reply:
security TF: would be better to be more specific and talk about tokens and OAuth specifically, as opposed to "credentials", which is confusing.

Reply:
Modified text as follows:
A WoT Thing may have credentials to establish its identity (authentication) and allow for the creation of trusted secure communication channels with other trusted and authenticated entities. Additional authorization information may be used in addition to manage access rights to specific entities.
Authorizations for secure M2M communication may also reveal the identity
of the accessor but this may not be desired when the accessing entity is a 
person or a device that can be associated with a person. In this case, mechanisms such as tokens and OAuth2 can be used, as with other web services, to avoid directly revealing user identities or requiring devices to associate authorizations directly with users.